validator/main.py

```diff
@@ -173,14 +183,31 @@ def validate(
             not self._disable_classifier and not self._disable_llm
         ):  # Use ensemble (Zero-Shot + Ensemble)
             applicable_topics = self.get_topics_ensemble(value, list(invalid_topics))
+            if not applicable_topics:
+                return PassResult()
         elif self._disable_classifier and not self._disable_llm:
-            # Use only LLM
-            applicable_topics = self.get_topics_llm(value, list(invalid_topics))
+            # Use only LLM, pass if good. If not, exit and fail
+            applicable_topics = asyncio.run(
+                self.get_topics_llm(value, list(invalid_topics))
+            )
+            if not applicable_topics:
+                return PassResult()
         else:
-            # Use only Zero-Shot
-            applicable_topics = self.get_topic_zero_shot(value, list(invalid_topics))
-
-        if not applicable_topics:
+            # Use only Zero-Shot, pass or fail here.
+            applicable_topics, scores = self.get_topic_zero_shot(
+                value, list(invalid_topics)
+            )
+            for topic, score in zip(applicable_topics, scores):
```

Comment:
This loop should apply to all configurations of `disable_llm` and `disable_classifier`, and not just Zero-shot classification only.